### ISSUE TITLE
Inputs with autocomplete="false" are reporting incorrect formatting in latest axe-core ruleset

### ISSUE THREAD

**Issue by 1Copenut** [usds/us-forms-system#316]
**Describe the bug**
The forms library inputs (selects, radio buttons, inputs) has a default `autocomplete="false"` attribute applied. The latest axe-core ruleset with WCAG 2.1 success criteria is treating this as an error: https://dequeuniversity.com/rules/axe/3.1/autocomplete-valid

**To Reproduce**
1. Install the [latest version of axe](https://www.deque.com/axe) on Chrome or Firefox
2. Navigate to a public form like https://www.va.gov/health-care/apply/application/veteran-information/personal-information
3. Right click to open Chrome or Firefox dev tools
4. Select `axe` from the dev tools top nav and click `Analyze`
5. Verify there should be 4 "autocomplete attribute must be used correctly" errors

**Expected behavior**
The autocomplete feature is one that's gotten a lot of interest in WCAG 2.1, and is being used to improve accessibility on common form patterns like credit card entry, addresses, and birthdays. To ensure good usability, the accessibility checker is now looking for a [value recognized by WCAG](https://www.w3.org/TR/WCAG21/#input-purposes), or no `autocomplete` at all. 

I verified that removing the autocomplete attribute completely from an input and running the scanner again suppressed the error.

I also verified that changing `autocomplete="false"` to `autocomplete="email"` on a text field asking for the user's email address suppressed the error.

**Screenshots**
![screen shot 2018-11-12 at 2 48 08 pm](https://user-images.githubusercontent.com/934879/48374284-034c2f80-e68a-11e8-9c04-7c5885a5534e.png)

---

**Potential Code Fix**
```js
const { autocompleteLabel, ... } = this.props;

<Input
  ...
  autocomplete={autocompleteLabel ? autocompleteLabel : null}
  ...
/>
```

**Desktop (please complete the following information):**
 - OS: MacOS Mojave
 - Browser Chrome latest
 - Version 70.0.3538.77

**Additional context**
* [Autocomplete attribute must be used correctly](https://dequeuniversity.com/rules/axe/3.1/autocomplete-valid)
* [WCAG 2.1 Input Purposes](https://www.w3.org/TR/WCAG21/#input-purposes)

**Comment by dmethvin-gov** [comment]
I had wondered why autocomplete was turned off (by using the invalid `"false"` value). I really like having my form filled in automatically when possible. @annekainicUSDS do you know if there was a specific reason, for example during testing people became confused about something that was autofilled? 

The spec for autofill token names is here: https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-detail-tokens

**Comment by annekainicUSDS** [comment]
I don't remember a specific reason this was turned off... so I think we can change it!

**Comment by dmethvin-gov** [comment]
Since the purpose of `autocomplete` seems to be to provide more details about the type of input it would make sense to have it be configurable. In general we can't know whether an arbitrary text input has special semantic meaning but the form author may be able to provide it. You should be able to set this via `"ui:options": { "autocomplete": "off" }` but I haven't tried it.

**Comment by 1Copenut** [comment]
@dmethvin-gov Agreed, there's no way to know ahead of time if an input will have special meaning and should be autofilled with previously remembered values.

Chrome does a fair job deriving autocomplete values using label text from what I've seen. Having the ability to explicitly turn off autocomplete, or set accepted values like `"ui:options": { "autocomplete": "email" }` should suppress the axe-core error without affecting user experience.

**Comment by rianfowler** [comment]
I've made a [PR](https://github.com/usds/us-forms-system/pull/324) to address this.

**Comment by rianfowler** [comment]
@1Copenut LMK if there are additional things we need to do in `vets-website` to address this issue. I'm tracking what we need to do here: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15969

**Comment by dmethvin-gov** [comment]
Should now be fixed by #324 (master) and #327 (1.x)
